plugins/modules/kea_command.py

```diff
@@ -0,0 +1,210 @@
+      - This module directly interfacees via UDS; the HTTP wrappers are not supported.
+    type: path
+    default: /run/kea/kea4-ctrl-socket
+extends_documentation_fragment:
+  - community.general.attributes
+  - community.general.attributes.platform
+attributes:
+  check_mode:
+    support: none
+  diff_mode:
+    support: none
+  platform:
+    support: full
+    platforms: posix
+"""
+
+EXAMPLES = r"""
+vars:
+  ipaddr: "192.168.123.45"
+  hwaddr: "00:00:5E:00:53:00"
+tasks:
+
+  # an example for a request acquiring information
+  - name: Get KEA DHCP6 status
+    kea_command:
+      command: status-get
+      rv_unchanged: [0]
+      socket: /run/kea/kea6-ctrl-socket
+    register: kea6_status
+  - name: Display registered status result
+    ansible.builtin.debug:
+      msg: KEA DHCP6 running on PID {{ kea6_status.response.arguments.pid }}
+
+  # an example for requests modifying state
+  - name: Remove existing leases for {{ ipaddr }}, if any
+    kea_command:
+      command: lease4-del
+      arguments:
+        ip-address: "{{ ipaddr }}"
+      rv_changed: [0]
+      rv_unchanged: [3]
+  - name: Add DHCP lease for {{ ipaddr }}
+    kea_command:
+      command: lease4-add
+      arguments:
+        ip-address: "{{ ipaddr }}"
+        hw-address: "{{ hwaddr }}"
+      rv_changed: [0]
+"""
+
+RETURN = r"""
+response:
+  description: The server JSON response.
+  returned: when available
+  type: dict
+"""
+
+import json
+import os
+import socket
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+# default buffer size for socket I/O
+BUFSIZ = 8192
+
+
+def _parse_constant(s):
+    raise ValueError(f'Invalid JSON: "{s}"')
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            command=dict(type="str", required=True),
+            arguments=dict(type="dict"),
+            rv_unchanged=dict(type="list", elements="int", default=[]),
+            rv_changed=dict(type="list", elements="int", default=[]),
+            socket=dict(type="path", default="/run/kea/kea4-ctrl-socket"),
+        ),
+    )
+
+    cmd = {}
+    cmd["command"] = module.params["command"]
+    if module.params["arguments"] is not None:
+        cmd["arguments"] = module.params["arguments"]
+    cmdstr = json.dumps(cmd, ensure_ascii=True, allow_nan=False, indent=None, separators=(",", ":"), sort_keys=True)
+    rvok = module.params["rv_unchanged"]
+    rvch = module.params["rv_changed"]
+    sockfn = module.params["socket"]
+
+    r = {"changed": False}
+    rsp = b""
+
+    if not os.path.exists(sockfn):
+        r["msg"] = f"socket ({sockfn}) does not exist"
+        module.fail_json(**r)
+
+    phase = "opening"
+    try:
+        with socket.socket(socket.AF_UNIX, socket.SOCK_STREAM) as sock:
+            phase = "connecting"
+            sock.connect(sockfn)
+            # better safe in case anything fails…
+            r["changed"] = True
```

Comment:
If the statement in line 165 below fails to send the command, then the module returns `changed=true` but nothing happened.